### ISSUE TITLE
Modifications to cookie functions causing createCookie to fail in IE7/IE8

### ISSUE THREAD

**Issue by rich97** [wojodesign/simplecart-js#118]
**Quoted from stackoverflow:**

I'm using simpleCart.js to setup a cart on the website. The problem is the cookies will not be created in IE7/IE8. So every page refresh we lose the items in the cart.

I've been looking at the code and it appears as though simpleCart uses the cookie functions from quirksmode

The only information I can find as to why it would be failing is here: http://www.dynamicdrive.com/forums/showthread.php?t=27197 something to do with how the cookies are set that IE doesn't like.

I modified the quirksmode function to look like this:

``` javascript
function createCookie(name,value,days) {
    if (days) {
        var date = new Date();
        date.setTime(date.getTime()+(days*24*60*60*1000));
        var expires = "; expires="+date.toGMTString();
    }
    else var expires = "";
    value = value.replace(/\=/g, '~');
    console.log(name + "=" + value + expires + "; path=/");
    document.cookie = name + "=" + escape(value) + expires + "; path=/";
    console.log(name + "=" + escape(value) + expires + "; path=/");
}
```

And here is the output:

```
sc_simpleCart_1=id~c2||name~British%20Curry%20Powder||price~3.5||quantity~4; expires=Thu, 18 Aug 2011 15:58:16 GMT; path=/
sc_simpleCart_1=id%7Ec2%7C%7Cname%7EBritish%2520Curry%2520Powder%7C%7Cprice%7E3.5%7C%7Cquantity%7E4; expires=Thu, 18 Aug 2011 15:58:16 GMT; path=/
```

I thought that maybe it was something to do with the space characters, but I replaced them with hyphens and it made no difference.

There is actually a live version of the site here: http://www.bitspicy.com/bit-spicy-shop.php try adding an item to the cart and refresh the page in IE8 to see the issue for yourself.

Many thanks, Rich.

**End Quote**

Basically it appears as though the use of escape() causes IE to go start crying like a spoilt child when trying to save the cookie using `document.cookie`.

One thing I was also curious about, why don't you use something like JSON rather than creating your own? JSON works very well with Javascript and I don't think it will make IE6/7/8 any more terrible than it already is.

**Comment by brettwejrowski** [comment]
Hey rich97, thanks for posting your issue.  To be honest, I believe there was a reason I didn't use json for storing values in a cookie originally... but I do not remember now what that reason would be (about 4 years ago), so I will take another look.

I haven't seen this issue before in IE, so I will check out your live site to see if I can figure out why escape() is pissing off IE.  There have also been some recent changed to the cookie storage functions, so that might be causing the new issues as well.

**Comment by rich97** [comment]
[Here is the JS for the site.](http://www.bitspicy.com/js/simpleCart.js) I didn't actually implement simpleCart on this site, I just added a feature (hacked in a discount code system.)

I'm fairly sure it's not to do with the discount hack as that info is saved in a separate cookie, which only stores an integer (as a string of course.) :P And there are no JS errors showing in IE.

You'll also notice that I reverted your cookie functions to the quirksmode methods just to get it working so you wont be able to see reproduce the error on that site. Sorry. :(

**Comment by kenlyle** [comment]
Hey all, 

I have a live site showing this issue at http://gilmanarts.com/portfolio/portfolio.html - I'd really like to solve this, but need a noob-resistant explanation.  Something like edit the simplecart.js file to ??? remove the use of esacape()
